server/tests/api_v4/authenticated/test_realm_create.py

```diff
@@ -24,23 +49,120 @@ async def test_authenticated_realm_create_ok(coolorg: CoolorgRpcClients, backend
         await spy.wait_event_occurred(
             EventRealmCertificate(
                 organization_id=coolorg.organization_id,
-                timestamp=timestamp,
-                realm_id=wksp_id,
+                timestamp=certif.timestamp,
+                realm_id=certif.realm_id,
                 user_id=coolorg.alice.device_id.user_id,
                 role_removed=False,
             )
         )
 
 
-# async def test_authenticated_realm_create_already_exists(
-#     coolorg: CoolorgRpcClients, backend: Backend
-# ) -> None:
-#     block_id = BlockID.new()
-#     block = b"<block content>"
+async def test_authenticated_realm_create_already_exists(
+    coolorg: CoolorgRpcClients, backend: Backend
+) -> None:
+    certif = realm_role_certificate(coolorg.alice, realm_id=coolorg.wksp1_id)
+    rep = await coolorg.alice.realm_create(
+        realm_role_certificate=certif.dump_and_sign(coolorg.alice.signing_key)
+    )
+    assert rep == authenticated_cmds.v4.realm_create.RepRealmAlreadyExists(
+        last_realm_certificate_timestamp=get_last_realm_certificate_timestamp(
+            testbed_template=coolorg.testbed_template,
+            realm_id=coolorg.wksp1_id,
+        )
+    )
+
+
+@pytest.mark.parametrize(
+    "kind",
+    (
+        "dummy_data",
+        "bad_author",
+        pytest.param(
+            "author_is_outsider",
+            marks=pytest.mark.xfail(
+                reason="TODO: It seems that outsider is allowed to create realm. Not implemented?"
+            ),
+        ),
```

Comment:
Not sure if this is OK or if the implementation is missing.

Reply:
This is actually expected behavior for outsider profile: realm_create is authorized, but not realm_share.